src/dkg_bipolynomial.erl

```diff
@@ -31,7 +35,8 @@ generate(Pairing, T) ->
                                                                   erlang:append_element(element(J+1, Acc2), RandCoeff2)),
                                                        erlang:append_element(element(I+1, Acc2), RandCoeff2))
                                     end, NewAcc, lists:seq(I+1, T))
-                end, list_to_tuple(lists:duplicate(T+1, {})), lists:seq(0, T)).
+                end, list_to_tuple(lists:duplicate(T+1, {})), lists:seq(0, T)),
```

Comment:
I was looking at this trying to figure out why you were doing the list_to_tuple conversions and back. I see that there's a setelement that sets entries in the tuple, but they're done one by one.. Why not collect them in an accumulator instead of setting tuple entries? Was that slower?

Reply:
This is mostly now a holdover from the original data structure. The construction is a bit tricky, so I didn't want to mess with it too much. Now that everything else is switched over we could redo it.